src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Embedded.cs

```diff
@@ -0,0 +1,341 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class AttributeTests_Embedded : CSharpTestBase
+    {
+        [Fact]
+        public void ReferencingEmbeddedAttributesFromADifferentAssemblyFails_Internal()
+        {
+            var reference = CreateCompilationWithMscorlib(@"
+[assembly:System.Runtime.CompilerServices.InternalsVisibleToAttribute(""Source"")]
+namespace Microsoft.CodeAnalysis
+{
+    internal class EmbeddedAttribute : System.Attribute { }
+}
+namespace TestReference
+{
+    [Microsoft.CodeAnalysis.Embedded]
+    internal class TestType1 { }
+
+    [Microsoft.CodeAnalysis.EmbeddedAttribute]
+    internal class TestType2 { }
+
+    internal class TestType3 { }
+}");
+
+            var code = @"
+class Program
+{
+    public static void Main()
+    {
+        var obj1 = new TestReference.TestType1();
+        var obj2 = new TestReference.TestType2();
+        var obj3 = new TestReference.TestType3(); // This should be fine
+    }
+}";
+
+            CreateCompilationWithMscorlib(code, references: new[] { reference.ToMetadataReference() }, assemblyName: "Source").VerifyDiagnostics(
+                // (6,38): error CS0234: The type or namespace name 'TestType1' does not exist in the namespace 'TestReference' (are you missing an assembly reference?)
+                //         var obj1 = new TestReference.TestType1();
+                Diagnostic(ErrorCode.ERR_DottedTypeNameNotFoundInNS, "TestType1").WithArguments("TestType1", "TestReference").WithLocation(6, 38),
+                // (7,38): error CS0234: The type or namespace name 'TestType2' does not exist in the namespace 'TestReference' (are you missing an assembly reference?)
+                //         var obj2 = new TestReference.TestType2();
+                Diagnostic(ErrorCode.ERR_DottedTypeNameNotFoundInNS, "TestType2").WithArguments("TestType2", "TestReference").WithLocation(7, 38));
+        }
+
+        [Fact]
+        public void ReferencingEmbeddedAttributesFromADifferentAssemblyFails_Public()
```

Comment:
>ReferencingEmbeddedAttributesFromADifferentAssemblyFails_Public [](start = 20, length = 63)

Should also test with an added module instead of an assembly reference. Should also test if both snippets are compiled into the same source module, there are no errors.